nucypher/blockchain/eth/actors.py

```diff
@@ -375,175 +434,152 @@ def perform_round_1(
 
-        # validate the active ritual tracker state
-        participant = self.coordinator_agent.get_participant_from_provider(
-            ritual_id=ritual_id, provider=self.checksum_address
+        ritual = self.coordinator_agent.get_ritual(
+            ritual_id=ritual_id,
+            transcripts=False,
         )
-        if participant.transcript:
-            self.log.debug(
-                f"Node {self.transacting_power.account} has already posted a transcript for ritual {ritual_id}; skipping execution"
-            )
-            return None
-
-        # check pending tx; since we check the coordinator contract
-        # above, we know this tx is pending
-        pending_tx = self.dkg_storage.get_transcript_receipt(ritual_id=ritual_id)
-        if pending_tx:
-            try:
-                txhash = pending_tx["transactionHash"]
-            except TypeError:
-                txhash = pending_tx
-            txhash = bytes(txhash).hex()
-            self.log.debug(
-                f"Node {self.transacting_power.account} has pending tx {txhash} "
-                f"for posting transcript for ritual {ritual_id}; skipping execution"
-            )
-            return None
 
         self.log.debug(
-            f"performing round 1 of DKG ritual #{ritual_id} from blocktime {timestamp} with authority {authority}."
+            f"performing round 1 of DKG ritual "
+            f"#{ritual_id} from blocktime {timestamp} "
+            f"with authority {authority}."
         )
 
-        # gather the cohort
-        ritual = self.coordinator_agent.get_ritual(ritual_id, with_participants=True)
-        nodes, transcripts = list(zip(*self._resolve_validators(ritual, ritual_id)))
-        nodes = sorted(nodes, key=lambda n: n.address)
-        if any(transcripts):
-            self.log.debug(
-                f"ritual #{ritual_id} is in progress {ritual.total_transcripts + 1}/{len(ritual.providers)}."
-            )
-
         # generate a transcript
+        validators = self._resolve_validators(ritual)
         try:
             transcript = self.ritual_power.generate_transcript(
-                nodes=nodes,
+                nodes=validators,
                 threshold=ritual.threshold,
                 shares=ritual.shares,
                 checksum_address=self.checksum_address,
-                ritual_id=ritual_id
+                ritual_id=ritual.id,
             )
         except Exception as e:
             # TODO: Handle this better #3096
-            self.log.debug(f"Failed to generate a transcript for ritual #{ritual_id}: {str(e)}")
+            self.log.critical(
+                f"Failed to generate a transcript for ritual #{ritual.id}: {str(e)}"
+            )
             raise e
 
         # store the transcript in the local cache
-        self.dkg_storage.store_transcript(ritual_id=ritual_id, transcript=transcript)
+        self.dkg_storage.store_transcript(ritual_id=ritual.id, transcript=transcript)
```

Comment:
@KPrasch I just realized that we don't use the stored transcript at all (Note that this is different from the hash for the transaction, it is the actual transcript). I'd like to remove this code (and analogous code for the aggregated transcript in round 2) - however, I'd prefer to do it in a follow-up PR.

Reply:
Filed https://github.com/nucypher/nucypher/issues/3426

Reply:
followed up: #3428